contracts/services/generic-ibc-transfer/src/msg.rs

```diff
@@ -0,0 +1,286 @@
+use std::collections::BTreeMap;
+
+use cosmwasm_schema::{cw_serde, QueryResponses};
+use cosmwasm_std::{Addr, CustomQuery, Deps, DepsMut, Uint128, Uint64};
+use cw_ownable::cw_ownable_query;
+use getset::{Getters, Setters};
+use valence_ibc_utils::types::PacketForwardMiddlewareConfig;
+use valence_macros::{valence_service_query, ValenceServiceInterface};
+use valence_service_utils::{
+    denoms::{CheckedDenom, UncheckedDenom},
+    error::ServiceError,
+    msg::ServiceConfigValidation,
+    ServiceAccountType,
+};
+
+#[cw_serde]
+pub enum ActionMsgs {
+    IbcTransfer {},
+}
+
+#[valence_service_query]
+#[cw_ownable_query]
+#[cw_serde]
+#[derive(QueryResponses)]
+/// Enum representing the different query messages that can be sent.
+pub enum QueryMsg {}
+
+#[cw_serde]
+#[derive(ValenceServiceInterface)]
+pub struct ServiceConfig {
+    pub input_addr: ServiceAccountType,
+    pub output_addr: String,
+    pub denom: UncheckedDenom,
+    pub amount: IbcTransferAmount,
+    pub memo: String,
+    pub remote_chain_info: RemoteChainInfo,
+    pub denom_to_pfm_map: BTreeMap<String, PacketForwardMiddlewareConfig>,
+}
+
+#[cw_serde]
+pub enum IbcTransferAmount {
+    FullAmount,
+    FixedAmount(Uint128),
+}
+
+#[cw_serde]
+pub struct RemoteChainInfo {
+    pub channel_id: String,
+    pub port_id: Option<String>,
```

Comment:
Extending on above, we shouldn't pass this parameter because it´s not being used.

Reply:
See my previous comment .. but I don't have a strong opinion about it. If you feel it's unnecessary, I can remove it.